CMakeLists.txt

```diff
@@ -91,7 +91,7 @@ target_link_libraries(img_output_u8_test PRIVATE Halide::Halide
                                                  ${CNPY}
                                                  img_output_u8)
 
-add_executable(sarbp_test sarbp.cpp PlatformData.cpp ImgPlane.cpp)
+add_executable(sarbp_test sarbp.cpp dft.cpp fftw.cpp PlatformData.cpp ImgPlane.cpp)
```

Comment:
Why two different source files?  The `fftw_plan` that we need for context is tightly coupled to the `call_dft` implementation.  Conceptually there could be need for other DFTs as well which would need their own plans, e.g., if we need to add support additional SAR functionality like autofocus.

I think we should just have `dft.h` and `dft.cpp`, or even `dft_extern.h` and `dft_extern.cpp` to provide more clarity on what these are for.  For example, I'm using a similar naming convention for `interp_extern` in a branch I still haven't merged - https://github.com/cimes-isi/halide-sar-app/tree/bp-ritsar-interp
Then the `fftw_plan` can still be local to the source file, and we add functions like `dft_ctx_init` and `dft_ctx_destroy` which manage this plan.

Reply:
Yes, just factoring call_dft out of main is what I started with (see commits in this PR), but then for some reason had to factor further. I just backtracked and re-tested with just the dft.cpp/h and it works fine for both default and CASPER. Re renaming -- can rename whenever.